rust/src/storage/s3.rs

```diff
@@ -13,6 +12,47 @@ use tokio::io::AsyncReadExt;
 
 use super::{parse_uri, ObjectMeta, StorageBackend, StorageError};
 
+enum ContinuationToken {
```

Comment:
I see what this code is doing, but I don't fully understand why. Would you mind adding some rustdoc @houqp

Reply:
I have moved this enum into the list_objs function so it only stays in the scope where it's being used. I have also added some doc string to expand on how it's being used in the object list code.